build.vsh

```diff
@@ -62,31 +62,42 @@ fn prepare_output_dir() {
 	os.mkdir('./bin') or { errorln('Failed to create output directory: ${err}') }
 }
 
+fn generate() bool {
+	os.chdir('./tree_sitter_v') or { return false }
+	if os.execute('npm run generate').exit_code == 0 {
+		os.chdir('../') or { return false }
+		println('${term.green('✓')} Run tree-sitter generate')
+		return true
+	}
+	return false
+}
+
 fn build(mode ReleaseMode, explicit_debug bool) {
 	println('Building v-analyzer at commit: ${build_commit}, build time: ${build_datetime} ...')
+	if generate() {
+		prepare_output_dir()
+		println('${term.green('✓')} Prepared output directory')
```

Comment:
It is unfortunately missing all error handling if the generate command fails. To give an example of adding error handling while also keeping the code simpler.

```suggestion
fn generate() ! {
	os.chdir('./tree_sitter_v')!
	os.execute_opt('npm run generate')!
	os.chdir('../')!
}

fn build(mode ReleaseMode, explicit_debug bool) {
	println('Building v-analyzer at commit: ${build_commit}, build time: ${build_datetime} ...')

	generate() or {
		eprintln('Failed to generate tree-sitter grammar. ${err}')
		exit(1)
	}
	println('${term.green('✓')} Generated tree-sitter grammar')
```

Also the code is not working directory independent.